pkg/agent/multicluster/mc_route_controller.go

```diff
@@ -329,13 +339,23 @@ func (c *MCRouteController) addMCFlowsForSingleCIImp(activeGW *mcv1alpha1.Gatewa
 		return errors.New("invalid Gateway IP")
 	}
 
+	var ciImportNoChange, activeGWNochange bool
 	if installedCIImp != nil {
 		oldTunnelPeerIPToRemoteGW := getPeerGatewayIP(installedCIImp.Spec)
-		if oldTunnelPeerIPToRemoteGW.Equal(tunnelPeerIPToRemoteGW) && installedCIImp.Spec.ServiceCIDR == ciImport.Spec.ServiceCIDR &&
-			sets.NewString(installedCIImp.Spec.PodCIDRs...).Equal(sets.NewString(ciImport.Spec.PodCIDRs...)) {
-			klog.V(2).InfoS("No difference between new and installed ClusterInfoImports, skip updating", "clusterinfoimport", ciImport.Name)
-			return nil
-		}
+		ciImportNoChange = oldTunnelPeerIPToRemoteGW.Equal(tunnelPeerIPToRemoteGW) && installedCIImp.Spec.ServiceCIDR == ciImport.Spec.ServiceCIDR &&
+			sets.NewString(installedCIImp.Spec.PodCIDRs...).Equal(sets.NewString(ciImport.Spec.PodCIDRs...))
+	}
+
+	if activeGW.Name == c.nodeConfig.Name {
```

Comment:
I feel easier to understand and more efficient if we check GW IP changed or not in syncMCFlowsForAllCIImps() and pass a boolean argument to this func. No strong opinion.

Reply:
You are right, it would be more efficient to move this kind of check out of a loop. fixed it. thanks.